core/src/ledger/storage_api/mod.rs

```diff
@@ -20,6 +20,18 @@ use crate::types::storage::{
     self, BlockHash, BlockHeight, Epoch, Header, TxIndex,
 };
 
+/// Write actions to help commit_block determine what parts of the storage to
+/// update
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum WriteActions {
```

Comment:
nit, perhaps better `WriteOpts`

Reply:
also, maybe this would be better as
```rust
 struct WriteOpts {
    pub modify_merkle_tree: bool,
    pub write_diffs: bool,
}

impl Default for WriteOpts {
    fn default() -> Self {
        Self { modify_merkle_tree: true, write_diffs: true }
    }
}
```

Reply:
even better we could use e.g. https://crates.io/crates/bitflags

Reply:
done in dd9d661621a428b60434aabba4c365fc9aa7597f